controller/linodeplacementgroup_controller.go

```diff
@@ -0,0 +1,321 @@
+
+	err = r.reconcileCreate(ctx, logger, pgScope)
+	if err != nil && !reconciler.HasConditionSeverity(pgScope.LinodePlacementGroup, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
+		logger.Info("re-queuing Placement Group creation")
+
+		res = ctrl.Result{RequeueAfter: reconciler.DefaultPGControllerReconcilerDelay}
+		err = nil
+	}
+
+	return
+}
+
+//nolint:dupl // same as VPC - future generics candidate.
+func (r *LinodePlacementGroupReconciler) reconcileCreate(ctx context.Context, logger logr.Logger, pgScope *scope.PlacementGroupScope) error {
+	logger.Info("creating placement group")
+
+	if err := pgScope.AddCredentialsRefFinalizer(ctx); err != nil {
+		logger.Error(err, "Failed to update credentials secret")
+
+		reconciler.RecordDecayingCondition(pgScope.LinodePlacementGroup, clusterv1.ReadyCondition, string(infrav1alpha2.CreatePlacementGroupError), err.Error(), reconciler.DefaultTimeout(r.ReconcileTimeout, reconciler.DefaultPGControllerReconcileTimeout))
+
+		r.Recorder.Event(pgScope.LinodePlacementGroup, corev1.EventTypeWarning, string(infrav1alpha2.CreatePlacementGroupError), err.Error())
+
+		return err
+	}
+
+	if err := r.reconcilePlacementGroup(ctx, pgScope, logger); err != nil {
+		logger.Error(err, "Failed to create Placement Group")
+
+		reconciler.RecordDecayingCondition(pgScope.LinodePlacementGroup, clusterv1.ReadyCondition, string(infrav1alpha2.CreatePlacementGroupError), err.Error(), reconciler.DefaultTimeout(r.ReconcileTimeout, reconciler.DefaultPGControllerReconcileTimeout))
+
+		r.Recorder.Event(pgScope.LinodePlacementGroup, corev1.EventTypeWarning, string(infrav1alpha2.CreatePlacementGroupError), err.Error())
+
+		return err
+	}
+	pgScope.LinodePlacementGroup.Status.Ready = true
+
+	if pgScope.LinodePlacementGroup.Spec.PGID != nil {
+		r.Recorder.Event(pgScope.LinodePlacementGroup, corev1.EventTypeNormal, "Created", fmt.Sprintf("Created Placement Group %d", *pgScope.LinodePlacementGroup.Spec.PGID))
+	}
+
+	return nil
+}
+
+//nolint:nestif,gocognit // As simple as possible.
+func (r *LinodePlacementGroupReconciler) reconcileDelete(ctx context.Context, logger logr.Logger, pgScope *scope.PlacementGroupScope) (ctrl.Result, error) {
+	logger.Info("deleting Placement Group")
+
+	if pgScope.LinodePlacementGroup.Spec.PGID != nil {
+		pg, err := pgScope.LinodeClient.GetPlacementGroup(ctx, *pgScope.LinodePlacementGroup.Spec.PGID)
+		if util.IgnoreLinodeAPIError(err, http.StatusNotFound) != nil {
+			logger.Error(err, "Failed to fetch Placement Group")
+
+			if pgScope.LinodePlacementGroup.ObjectMeta.DeletionTimestamp.Add(reconciler.DefaultTimeout(r.ReconcileTimeout, reconciler.DefaultPGControllerReconcileTimeout)).After(time.Now()) {
+				logger.Info("re-queuing Placement Group deletion")
+
+				return ctrl.Result{RequeueAfter: reconciler.DefaultPGControllerReconcilerDelay}, nil
+			}
+
+			return ctrl.Result{}, err
+		}
+
+		if pg != nil {
+			if len(pg.Members) > 0 {
```

Comment:
could we potentially move this into it's own function to remove the `nestif` exception?